src/internal/ppsutil/util.go

```diff
@@ -431,7 +431,7 @@ func ContainsS3Inputs(in *pps.Input) bool {
 // sidecar server) and the worker (which passes the endpoint to the user code)
 // need to know it.
 func SidecarS3GatewayService(jobID string) string {
-	return "s3-" + jobID
+	return "s3-" + strings.Replace(strings.ToLower(jobID), "@", "-", 1)
```

Comment:
feels kind of unnecessary to build a string then modify it. Also jobID is kind of ambiguous now, I would prefer either passing in a job proto or two strings

Reply:
that makes sense. Is the GUID in the ID the commit set ID?